GameServerLib/GameObjects/Missiles/Projectile.cs

```diff
@@ -289,27 +289,34 @@ protected bool CheckIfValidTarget(IAttackableUnit unit)
             {
                 // TODO: Verify all
                 // Order is important
-                case ILaneMinion _ when ((SpellData.Flags & (int)SpellFlag.SPELL_FLAG_AFFECT_MINIONS) > 0):
+                case ILaneMinion _ when SpellData.Flags.HasFlag(SpellDataFlags.AffectMinions)
+                                    && !SpellData.Flags.HasFlag(SpellDataFlags.IgnoreLaneMinion):
                     return true;
-                case IMinion m when (!m.IsPet && ((SpellData.Flags & (int)SpellFlag.SPELL_FLAG_AFFECT_NOT_PET) > 0))
-                                || (m.IsPet && ((SpellData.Flags & (int)SpellFlag.SPELL_FLAG_AFFECT_USEABLE) > 0))
-                                || (m.IsWard && ((SpellData.Flags & (int)SpellFlag.SPELL_FLAG_AFFECT_WARDS) > 0))
-                                || (!m.IsClone && ((SpellData.Flags & (int)(SpellFlag.SPELL_FLAG_IGNORE_CLONES - 1)) > 0))
-                                || (SpellData.Flags & (int)SpellFlag.SPELL_FLAG_AFFECT_ALL_UNIT_TYPES) > 0:
+                case IMinion m when (!m.IsPet && SpellData.Flags.HasFlag(SpellDataFlags.AffectNotPet))
+                                || (m.IsPet && SpellData.Flags.HasFlag(SpellDataFlags.AffectUseable))
+                                || (m.IsWard && SpellData.Flags.HasFlag(SpellDataFlags.AffectWards))
+                                || (!m.IsClone && SpellData.Flags.HasFlag(SpellDataFlags.IgnoreClones))
```

Comment:
The previous code uses `(int)(SpellFlag.SPELL_FLAG_IGNORE_CLONES - 1)` and for some reason here there isn't this -1.

Reply:
Because the previous SpellFlag enum was using hex for its value, and the comparison was between int values, it had to subtract 1 because IgnoreClones is 0x80000000, which is -int.MaxValue, which is less than 0.

Reply:
The new method ignores the int comparison by using HasFlag.